subsys/shell/shell.c

```diff
@@ -1238,11 +1238,8 @@ int shell_uninit(const struct shell *shell)
 	__ASSERT_NO_MSG(shell);
 
 	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
-		struct k_poll_signal *signal =
-				&shell->ctx->signals[SHELL_SIGNAL_KILL];
-
-		/* signal kill message */
-		(void)k_poll_signal_raise(signal, 0);
+		instance_uninit(shell);
```

Comment:
this may not work if called from shell thread context. Uninit is done but shell thread will continue to execute until hitting `k_poll` in `shell_thread`. Is it necessary in this PR to fix the issue? If not, please remove it. `shell_uninit` requires more work and dedicated tests.

Reply:
I think I can move this to another issue and track it there.

Reply:
The issue I am trying to resolve here is preventing deadlock if tx_ready is blocking.  This occurs when the USB_CDC is disconnected.

Reply:
I believe the problem is because the log backend remains enabled and then blocks when a log message is processed.  The log backend is enabled when the shell thread starts and the existing method to stop the backend it involves using instance_uninit() which calls shell_log_backend_disable().  This is why I was using shell_uninit() and encountered this issue.  I originally considered shell_stop() but that leaves the backend running.

I looked a little closer and perhaps the issue is best resolved by enabling and disabling the backend using shell_start and shell_stop respectively?  It occurs to me that if you "stop" the shell you would want to stop the log backend as well?

Reply:
I ran an experiment where I enable and disable the log backend using shell_start() and shell_stop() and this has resolved the issue where the shell deadlocks when the backend is left running.  I will push a commit so you can take a closer look.